src/main/java/org/assertj/core/api/AbstractAssert.java

```diff
@@ -1094,4 +1091,11 @@ private void propagateAssertionInfoFrom(AbstractAssert<?, ?> assertInstance) {
     return (ASSERT) assertFactory.createAssert(extractedValue).withAssertionState(myself);
   }
 
+  @CheckReturnValue
+  public SELF ignoreWhen(Predicate<ACTUAL> predicate) {
+    requireNonNull(predicate, shouldNotBeNull("predicate")::create);
```

Comment:
`requireNonNull` with a `Supplier` is indeed what we should use in many other places! 👍